lib/providers/messages_provider.dart

```diff
@@ -1,112 +1,89 @@
-import 'package:flutter/material.dart';
+import 'dart:async';
+
 import 'package:graphql_flutter/graphql_flutter.dart';
 import 'package:mm_flutter_app/__generated/schema/operations_message.graphql.dart';
 import 'package:mm_flutter_app/__generated/schema/schema.graphql.dart';
+import 'package:mm_flutter_app/utilities/errors/crash_handler.dart';
 
 import 'base/base_provider.dart';
 import 'base/operation_result.dart';
 
 typedef ChannelMessage = Query$FindChannelMessages$findChannelMessages;
+typedef FilteredChannelMessage
+    = Query$FindChannelMessagesWithOptions$findChannelMessages;
 typedef UnseenMessage
     = Query$InboxUnseenMessages$myInbox$channels$unseenMessages;
 
 class MessagesProvider extends BaseProvider {
-  MessagesProvider({required super.client}) {
-    // subscribeToMessages();
-  }
-
-  _addMessageToCache({cache, required String channelId, message}) {
-    final req = QueryOptions(
-      document: documentNodeQueryFindChannelMessages,
-      variables: Variables$Query$FindChannelMessages(
-        filter: Input$ChannelMessageListFilter(
-          channelId: channelId,
-        ),
-      ).toJson(),
-    ).asRequest;
-    final response = cache.readQuery(req);
-
-    // debugPrint('mutation result');
-    // print(result);
-
-    // debugPrint('Channels cache response');
-    // debugPrint('Channels in cache: ${channelsData.length}');
-    response?['findChannelMessages'].add(message);
+  MessagesProvider({required super.client});
 
-    // print(response?['messages']);
-
-    if (response != null) {
-      cache.writeQuery(
-        req,
-        broadcast: true,
-        data: response,
-      );
-    }
+  // Queries
+  Future<OperationResult<List<ChannelMessage>>> findChannelMessages({
+    required Input$ChannelMessageListFilter input,
+    bool fetchFromNetworkOnly = false,
```

Comment:
Good job using the FetchPolicy, I got curious why not using only the FetchPolicy to control whether getting from network or not?

Would cacheAndNetwork policy give the same result? (https://github.com/zino-hofmann/graphql-flutter/blob/main/packages/graphql/lib/src/core/policies.dart#L27)

Reply:
cacheAndNetwork has to be used carefully because the returned data will always be stale. There are cases when this is ok, but chat is probably the one exception that always needs the latest data.